### ISSUE TITLE
Remove the `--packaging-format` option for Linux System packaging

### ISSUE THREAD

**Issue by rmartin16** [beeware/briefcase#1385]
### Describe the bug

While it may be possible to build packages that are not native to the distro, doing this in practice does not work.

Even if you build a deb in Fedora, you can't install it in to another Fedora environment; none of the dependent packages will be available.

Building a pkg with `makepkg` requires all the dependencies to already be installed...but how can I install them if I'm not on Arch?

### Steps to reproduce

Attempt to build a package that is not native to the target.

### Expected behavior

The packaging format is chosen based on the target distro....which is the default (and sane) behavior anyway.

### Environment

- Operating System: fedora 37
- Python version: 3.11.4
- Software versions:
  - Briefcase: `0.3.16.dev58+g5483fd9a`

**Comment by freakboy3742** [comment]
Agreed that it shouldn't be possible to manually specify `rpm` packaging on Arch, etc. 

There are 2 caveats on this:

1. We probably want to retain the `packaging_format` option to support source packaging (#1152). I guess that means the future state allows for 2 packaging formats - "binary" and "source" - which are then converted into "rpm" or "srpm" (et al) based on local platform conditions.

2. One of the aims of allowing `--packaging_format=rpm` is to allow RPM packaging when the vendor base cannot be determined. If you've got an RPM-based distro that Briefcase can't auto-identify, it is desirable to still be able to build an RPM package.

Whether (2) is useful enough in practice is an open question. Linux distros have a very long tail, but (a) the intersection of niche Linux distros and Briefcase users is likely very small, and (b) I'm not sure how much we want to take on the burden of claiming support for niche distros when we have limited ability (or interest) in actively testing that support.

If we were do make this change, my immediate impression is that we would:

1. Drop the platform-specific formats from the packaging format list in `LinuxSystemPackageCommand`
2. Modify `package.py` so that the `app.packaging_format = packaging_format` assignment is done by a method that the `LinuxSystemPackageCommand` can override that method, doing what `verify_app_tools()` is doing at present.
